qa/tasks/cephfs/test_nfs.py

```diff
@@ -876,3 +891,55 @@ def test_nfs_export_creation_at_symlink(self):
                 raise
         self.ctx.cluster.run(args=['rm', '-rf', f'{mnt_pt}/*'])
         self._delete_cluster_with_fs(self.fs_name, mnt_pt, preserve_mode)
+
+    def test_cephfs_export_update_with_nonexistent_dir(self):
+        """
+        Test that invalid path is not allowed while updating a CephFS
+        export.
+        """
+        self._create_cluster_with_fs(self.fs_name)
+        self._create_export(export_id=1)
+
+        try:
+            self.update_export(self.cluster_id, "/not_existent_dir",
+                               self.pseudo_path, self.fs_name)
+        except CommandFailedError as e:
+            if e.exitstatus != errno.ENOENT:
+                raise
+
+        self._delete_export()
+        self._delete_cluster_with_fs(self.fs_name)
+
+    def test_cephfs_export_update_at_non_dir_path(self):
+        """
+        Test that non-directory path are not allowed while updating a CephFS
+        export.
+        """
+        mnt_pt = '/mnt'
+        preserve_mode = self._sys_cmd(['stat', '-c', '%a', mnt_pt])
+        self._create_cluster_with_fs(self.fs_name, mnt_pt)
+        self.ctx.cluster.run(args=['touch', f'{mnt_pt}/testfile'])
+        self._create_export(export_id=1)
+
+        # test at a file path
+        try:
+            self.update_export(self.cluster_id, "/testfile",
+                               self.pseudo_path, self.fs_name)
+        except CommandFailedError as e:
+            if e.exitstatus != errno.ENOTDIR:
+                raise
+
+        # test at a symlink path
+        self.ctx.cluster.run(args=['mkdir', f'{mnt_pt}/testdir'])
+        self.ctx.cluster.run(args=['ln', '-s', f'{mnt_pt}/testdir',
+                                   f'{mnt_pt}/testdir_symlink'])
+        try:
+            self.update_export(self.cluster_id, "/testdir_symlink",
+                               self.pseudo_path, self.fs_name)
+        except CommandFailedError as e:
+            if e.exitstatus != errno.ENOTDIR:
```

Comment:
@vshankar @phlogistonjohn the only thing pending in this PR is the problem we discussed above i.e. to make sure we report `ENOTDIR` (and not the generic `EIO`) here in our case since this is a single export update, there is a PR I raised for this https://github.com/ceph/ceph/pull/53240; it does the job but very badly and I feel it is not a good solution because it doesn't return the err json response and it just returns the errno and the big traceback(which looks really bad). So things needs to be addressed at root level i.e. in `object_format.py`'s `Responder` class. I had a private conversation with John regarding this on Friday and we are going to discuss this in detail in Tueday's orch weekly meeting. So my question is should we wait until we come to a proper solution for the errno or just merge this and make the changes to the test case later on? because this is quite an important fix I'd vouch for getting this merged.